src/main/java/org/wildfly/naming/client/util/NamingUtils.java

```diff
@@ -62,6 +62,23 @@ public static CompositeName toCompositeName(final Name name) throws InvalidNameE
         }
     }
 
+    /**
+     * Create a CompositeName where each name segment is equal to the name segment in the source name.
+     *
+     * @param name the source name
+     * @return a {@link CompositeName} where each name segment is equal to the name segment in the source name
+     * @throws InvalidNameException if an error occurs while converting the source name to a {@link CompositeName}
+     */
+    public static CompositeName toDecomposedCompositeName(final Name name) throws InvalidNameException {
+        if (name instanceof CompositeName) {
+            return (CompositeName) name;
+        } else {
+            final CompositeName compositeName = new CompositeName();
+            compositeName.addAll(new CompositeName(name.toString()));
```

Comment:
I think this won't work right because the name might be a compound name using a different scheme than the CompositeName parser can support.  I think the source name components have to be added one by one to the composite name to function correctly for name types that work differently than SimpleName (i.e. maybe right-to-left or using a non-slash separator).